pl_bolts/models/rl/dqn_model.py

```diff
@@ -242,29 +270,45 @@ def configure_optimizers(self) -> List[Optimizer]:
         optimizer = optim.Adam(self.net.parameters(), lr=self.lr)
         return [optimizer]
 
-    def prepare_data(self) -> None:
+    def _dataloader(self) -> DataLoader:
         """Initialize the Replay Buffer dataset used for retrieving experiences"""
-        self.source = ExperienceSource(self.env, self.agent)
         self.buffer = ReplayBuffer(self.replay_size)
         self.populate(self.warm_start_size)
 
-        self.dataset = RLDataset(self.buffer, self.sample_len)
+        self.dataset = ExperienceSourceDataset(self.train_batch)
+        return DataLoader(dataset=self.dataset, batch_size=self.batch_size)
 
     def train_dataloader(self) -> DataLoader:
         """Get train loader"""
-        return DataLoader(dataset=self.dataset, batch_size=self.batch_size)
+        return self._dataloader()
 
     def test_dataloader(self) -> DataLoader:
         """Get test loader"""
         return DataLoader(dataset=self.dataset, batch_size=self.batch_size)
 
     @staticmethod
-    def add_model_specific_args(arg_parser: argparse.ArgumentParser) -> argparse.ArgumentParser:
+    def make_env(env_name: str, seed: int) -> gym.Env:
```

Comment:
```suggestion
    def make_ environment(env_name: str, seed: int) -> gym.Env:
```
is create/prepare better word?

Reply:
i think make_environment is fine, but Im ok with changing it